.github/workflows/push.yml

```
@@ -0,0 +1,13 @@
+name: Pre-commit hooks
+
+on: [push]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+      - uses: pre-commit/action@v3.0.1
```

I know this is coming from the official example, but if you do not intended to update the patch version periodically, I believe not having it pinned is better.

```suggestion
      - uses: pre-commit/action@v3
```